[Joshua Gan Yi En] iP

dragonesejosh opened the pull request:
### Yale ChatBot
> Don't download it. ─ Stellar review from one (**un**)satisfied customer.

Current Features:

1. Unoriginal personality
2. ~Some~ Too much OOP
3. _Basic_ task management

Other (Upcoming) Features:

- [ ] May or may not add personality
- [ ] Bare-bones GUI

Why download it?

- To easily manage your tasks ❌
- Free from bugs ❌
- Headaches from horrible UI ✅
- Most bland ChatBot known to man ✅

To keep your sanity, add this line of code to the `main` method:

``` java
public static void main(String[] args) {
    // ADD THIS LINE
    return;
    // Make sure this code never sees the light of day
    new Yale(TASKS_FILE).run();
}
```

To make a far better ChatBot yourself, here's the [original](https://github.com/nus-cs2103-AY2425S2/ip).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

dragonesejosh committed:
Added basic functionality to chatbot
Changed name from Duke to Yale

dragonesejosh committed:
Added echoing to the chatbot
Added indents to chatbot output

dragonesejosh committed:
Added ability to store text and list them all out when asked

dragonesejosh committed:
Converted tasks to new Task class

dragonesejosh committed:
Added task marking and unmarking

dragonesejosh committed:
Added three new task types

dragonesejosh committed:
Added ability to add different tasks

dragonesejosh committed:
Added automated testing

dragonesejosh committed:
Added error handling

dragonesejosh committed:
Added ability to delete tasks

dragonesejosh committed:
Updated automated testing

dragonesejosh committed:
Add reading/writing of tasks from/to file
Update ./gitignore to include tasks file

dragonesejosh committed:
Update runtest.bat to remove tasks file before running

dragonesejosh committed:
Merge branch 'branch-Level-7'

dragonesejosh committed:
Add basic storing of dates for Deadline and Event

dragonesejosh committed:
Update test cases to test for date recognition

dragonesejosh committed:
Merge branch 'branch-Level-8'

dragonesejosh committed:
Add new classes: Ui, Storage, Parser, TaskList

dragonesejosh committed:
Implement Ui

dragonesejosh committed:
Add Task.java
Move Task and subclasses to Task.java

dragonesejosh committed:
Implement Storage

dragonesejosh committed:
Implement TaskList

dragonesejosh committed:
Implement Parser

dragonesejosh committed:
Add Command.java
Move command checking and running to Command

dragonesejosh committed:
Merge branch 'branch-A-MoreOOP'

dragonesejosh committed:
Move all classes to yale package

dragonesejosh committed:
Merge branch 'branch-A-Packages'

dragonesejosh committed:
Merge remote-tracking branch 'origin/add-gradle-support'

dragonesejosh committed:
Enable Gradle to build and run application
Fix bug that caused "list" command to exit application
Remove "ERROR" from start of printError since Gradle stops the program

dragonesejosh committed:
Merge branch 'branch-A-Gradle'

dragonesejosh committed:
Add JUnit tests for Command and Parser
Add dummy instances for several classes to use in testing

dragonesejosh committed:
Merge branch 'branch-A-JUnit'

dragonesejosh committed:
Change import order to follow coding standard

dragonesejosh committed:
Add JavaDoc comments to most public methods and some public classes

dragonesejosh committed:
Add command to find tasks by keywords

dragonesejosh committed:
Merge branch 'branch-A-CodingStandard'

dragonesejosh committed:
Merge branch 'branch-A-JavaDoc'

dragonesejosh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/yale/TaskList.java

dragonesejosh committed:
Add checkstyle functionality

dragonesejosh committed:
Edit checkstyle.xml to suppress indent warning for lambda-style switch statements

dragonesejosh committed:
Merge branch 'branch-A-CheckStyle'

dragonesejosh committed:
Change UI from text to GUI
Fix bug with delete error checking
Fix bug with read tasks not being marked done

dragonesejosh committed:
Merge branch 'branch-Level-10'

Meatsushi64 commented on src/main/java/yale/Parser.java:
```diff
@@ -0,0 +1,72 @@
+     * @param msg The input string to parse.
+     * @return false if the program should halt, true otherwise.
+     */
+    public void parseMsg(String msg) {
+        if (msg.equals("bye")) {
+            ui.goodbye();
+            return;
+        } else if (msg.equals("list")) {
+            taskList.listOut();
+            return;
+        }
+        for (Command cmd : COMMANDS) {
+            if (cmd.tryCommand(taskList, storage, msg)) {
+                storage.writeTasks(taskList.getTasks());
+                return;
+            }
+        }
+        tryFindCommand(msg);
+    }
```
this is an interesting implementation. why did you choose to try the command through all the messages, instead of using a case switch using an enum of COMMANDS?

Meatsushi64 commented on src/main/java/yale/Storage.java:
```diff
@@ -0,0 +1,52 @@
+                    tasks.add(task);
+                }
+            }
+        } catch (Exception ignored) {}
```
```suggestion
        } catch (Exception ignored) {
            throw new YaleException(ignored.getMessage);
        }
```

i think it would be good to throw an exception to surface up to the Yale bot so that the user can see that there is a problem reading the file if any

but between you and me, yolo

Meatsushi64 commented on src/main/java/yale/Task.java:
```diff
@@ -0,0 +1,130 @@
+        this.isDone = done;
+    }
+
+    public String toString() {
+        return "[%s] %s".formatted(isDone ? "X" : " ", name);
+    }
```
massive big brain to put the isDone within the toString() instead of making a whole new method 🆙 🔝

Meatsushi64 reviewed (commented):
Overall love the memes in your PR description. 

I have no idea what it is with the UI stuff but I guess I'll find out soon...

Maybe would be good to surface any exception if any to the user

cheers man 👍

dragonesejosh committed:
Add title to GUI

dragonesejosh committed:
Merge branch 'branch-Level-10'

dragonesejosh committed:
Fix bug with writing to storage twice

dragonesejosh committed:
Add Stream functionality

dragonesejosh committed:
Add assertions

ernestlsy commented on src/main/java/yale/Parser.java:
```diff
@@ -0,0 +1,72 @@
+            new Command("delete", "delete [id]", "delete (\\d+)",
+                    (t, m) -> t.deleteTask(Integer.parseInt(m.group(1)))),
+            new Command("find", "find [keyword]", "find (.+)",
+                    (t, m) -> t.listSearch(m.group(1)))
```
Should these lines be indented by 4 spaces instead?

ernestlsy commented on src/main/java/yale/Storage.java:
```diff
@@ -0,0 +1,52 @@
+     */
+    public ArrayList<Task> readTasks() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try (Scanner in = new Scanner(file)) {
```
Nice use of try-with-resources statement to handle closing of the Scanner!

ernestlsy commented on src/main/java/yale/Ui.java:
```diff
@@ -0,0 +1,63 @@
+    /**
+     * Creates a Ui to display output.
+     */
+    public Ui() {}
```
Declaring this default constructor seems unnecessary.

ernestlsy reviewed (commented):
LGTM! No significant issues besides a few nits

dragonesejosh committed:
Handle ignored catch blocks in Storage

Previously the read and write methods of Storage caught
general Exceptions, changed it to catch only FileNotFoundException
and handle it accordingly.

dragonesejosh committed:
Add Ui param to Storage creation

The previous commit changed the Storage constructor,
but the creation of the Storage object was not updated
to reflect that.

Improve logic flow of Commands and Parser

The bye and list commands have been added to the list
of Commands, and are no longer in their own separate
if else branch. Furthermore, the partialMatch method has
been removed in favour of simply checking in tryCommand,
since it is unnecessary.

dragonesejosh committed:
Merge pull request #1 from dragonesejosh/branch-A-Streams

Add Stream functionality

dragonesejosh committed:
Merge pull request #2 from dragonesejosh/branch-A-Assertions

Add assertions

dragonesejosh committed:
Merge branch 'master' into branch-A-CodeQuality

dragonesejosh committed:
Merge pull request #3 from dragonesejosh/branch-A-CodeQuality

Branch a code quality

dragonesejosh committed:
Remove max width restriction
Add help command

dragonesejosh committed:
Merge branch 'branch-C-Help'

dragonesejosh committed:
Simplify Command logic for commands with no params

dragonesejosh committed:
Merge branch 'branch-C-Help'

dragonesejosh committed:
Add basic user guide

dragonesejosh committed:
Merge branch 'branch-A-UserGuide'

dragonesejosh committed:
Add example usage of commands to user guide

dragonesejosh committed:
Merge branch 'branch-A-UserGuide'

dragonesejosh committed:
Remove table which did not display properly on page

dragonesejosh committed:
Merge branch 'branch-A-UserGuide'

dragonesejosh committed:
Update tests to be consistent with changed classes

Remove empty constructor in Ui
